tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -471,14 +480,19 @@ SpanBuffer SwapBuffers()
                 // One buffer is full, request an eager flush
                 RequestFlush();
 
-                if (buffer.TryWrite(spans, ref _temporaryBuffer, chunkSamplingPriority))
+                if (buffer.TryWrite(spans, ref _temporaryBuffer, chunkSamplingPriority) == SpanBuffer.WriteStatus.Success)
                 {
                     // Serialization to the secondary buffer succeeded
                     return;
                 }
             }
 
             // All the buffers are full :( drop the trace
+            DropTrace(spans);
+        }
+
+        private void DropTrace(ArraySegment<Span> spans)
+        {
             Interlocked.Increment(ref _droppedSpans);
```

Comment:
Should this increment by `spans.Length`?

Reply:
The metric counts the number of times we dropped spans, rather than the number of spans. I agree the name is confusing:

```
                    var droppedSpans = Interlocked.Exchange(ref _droppedSpans, 0);

                    if (droppedSpans > 0)
                    {
                        Log.Warning("{Count} traces were dropped since the last flush operation.", droppedSpans);
                    }
```

Reply:
Yeah... `_droppedSpans` really needs renaming to `_droppedTraces`